packages/plugin-react/src/index.ts

```diff
@@ -368,3 +391,25 @@ viteReact.preambleCode = preambleCode
 function loadPlugin(path: string): Promise<any> {
   return import(path).then((module) => module.default || module)
 }
+
+function createBabelOptions(
+  file: string,
+  rawOptions?: BabelOptions,
+  ssr?: boolean
+) {
+  const babelOptions = {
+    ssr,
+    file,
```

Comment:
Another option is to use `Object.defineProperties` to define `ssr` and `file` options with `enumerable: false`. That should stop Babel from tripping on them. 🤔

Reply:
I feel the intention would be less obvious and require a comment. But I can do that if you feel it'd be better.

Reply:
Why are these in the same object? I think the API would be more clear with three args. ssr and file are quite different than the rest of the object
```
type ReactBabelHook = (
  babelConfig: ReactBabelOptions,
  options: { ssr, file }, // or context?
  config: ResolvedConfig
) => void
```

Reply:
ok let's do that, and I vote `context` as the name

Reply:
Done.

Reply:
Thanks for the changes @cyco130!

Sorry, one last question 👀
Why is it called `file`? Now that it is in the second parameter, should we rename it to `id`? file isn't that good in case there are virtual modules, no?

Reply:
Fixed (wasn't my idea anyway :P )